Application.cpp

```diff
@@ -315,11 +316,12 @@ bool Application::onInit() {
 	bindGroupLayoutDesc.entryCount = (uint32_t)m_bindingLayoutEntries.size();
 	bindGroupLayoutDesc.entries = m_bindingLayoutEntries.data();
 	BindGroupLayout bindGroupLayout = m_device.createBindGroupLayout(bindGroupLayoutDesc);
+	auto wgpu_bindGroupLayout = WGPUBindGroupLayout(bindGroupLayout);
 
 	// Create the pipeline layout
 	PipelineLayoutDescriptor layoutDesc{};
 	layoutDesc.bindGroupLayoutCount = 1;
-	layoutDesc.bindGroupLayouts = &(WGPUBindGroupLayout)bindGroupLayout;
+	layoutDesc.bindGroupLayouts = &wgpu_bindGroupLayout;
```

Comment:
The correct fix is `(WGPUBindGroupLayout*)&bindGroupLayout;` (dereferencing before casting rather than the other way around)

Reply:
It appears the classes are not related by inheritance (at least that's what the compiler told me, actually can't find where `WGPUBindGroupLayoutImpl` is defined).
The compiler error I got was that it can't take the address of an rvalue, i.e., a temporary `WGPUBindGroupLayout` was created with the original code.

Reply:
`(WGPUBindGroupLayout)bindGroupLayout` is the rvalue that clang has trouble getting the address of, but if you swap `&` and `(cast)` you no longer have the problem: `&bindGroupLayout` is well defined, and casting to `WGPUBindGroupLayout*` is a pointer cast that no compiler should complain about. If it does nonetheless, try `reinterpret_cast<WGPUBindGroupLayout*>(&bindGroupLayout)`.

NB: `WGPUBindGroupLayoutImpl` is a Dawn thing that should not affect the way one uses the API. And indeed the `wgpu::BindGroupLayout` provided by the wrapper does not inherit from the standard `WGPUBindGroupLayout` but rather wraps it, hence the need for a manual cast here unfortunately.

Reply:
Just checked the HANDLE macro, as you did implement a cast operator to W&, the following also works and does not require an ugly reinterpret_cast:
`	layoutDesc.bindGroupLayouts = &static_cast<WGPUBindGroupLayout&>(bindGroupLayout);`